nemo/collections/llm/gpt/model/megatron/hyena/hyena_layer_specs.py

```diff
@@ -49,9 +57,39 @@ def _raise_te_import_error(*args, **kwargs):
     TERowParallelLinear = _raise_te_import_error
     TEDotProductAttention = _raise_te_import_error
 
-# Layer spec with TE modules
-if HAVE_TE:
-    hyena_stack_spec = ModuleSpec(
+
+def get_hyena_stack_spec(use_te=HAVE_TE, vortex_style_fp8=False):
+    if use_te:
+        row_linear = TERowParallelLinear
+        col_linear = TELayerNormColumnParallelLinear
+        pre_layernorm = IdentityOp # fused Norm+Linear, so no pre norm
+        core_attention=TEDotProductAttention
+    else:
+        row_linear = RowParallelLinear
+        col_linear = ColumnParallelLinear
+        pre_layernorm = TENorm # would raise error if attempt to use
+        core_attention=DotProductAttention
+    dense_projection = col_linear
+
+    import torch
+    if torch.is_inference_mode_enabled():
+        from megatron.core.post_training.modelopt.layers import Linear
+        row_linear = Linear
+        col_linear = RMSNormLinear
+
+        if vortex_style_fp8:
+            dense_projection=RMSNormTELinearWithFp8Padding
+
+            alt = TELayerNormColumnParallelLinearWithFp8Padding
+            from os import getenv
+            if getenv("ALT_HYENA_DENSE_PROJECTION", "") == alt.__name__:
+                import logging as log
+                log.warning(f"Using {alt.__name__}")
+                dense_projection=alt
+        else:
+            dense_projection=TELayerNormColumnParallelLinear
```

Comment:
won't this inactivate tensor parallelism (or do something weird with it) if the user happens to use an `torch.inference_mode()` context for any reason? This seems like sneaky behavior. Don't we probably want to have the option of doing inference with TP enabled for bigger models?